src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```diff
@@ -51,13 +52,22 @@ public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
      */
     public AddressBook toModelType() throws IllegalValueException {
         AddressBook addressBook = new AddressBook();
+
         for (JsonAdaptedPerson jsonAdaptedPerson : persons) {
             Person person = jsonAdaptedPerson.toModelType();
             if (addressBook.hasPerson(person)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_PERSON);
             }
             addressBook.addPerson(person);
         }
+
+        // Convert clients to model type before bookings
+        // As bookings are dependent on clients
+        for (JsonAdaptedBooking jsonAdaptedBooking : bookings) {
+            Booking booking = jsonAdaptedBooking.toModelType(addressBook);
+            addressBook.addBooking(booking);
```

Comment:
Should a TODO be added here to check for duplicate bookings in the future?

Reply:
I think it's not necessary because I have it listed as a task on the gantt chart. It may be better not to put TODOs around the codebase without any one in charge of them.